packages/medusa/src/api-v2/store/products/helpers.ts

```diff
@@ -0,0 +1,97 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  MedusaContainer,
+  MedusaPricingContext,
+  PricingTypes,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  isPresent,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import { refetchEntity } from "../../utils/refetch-entity"
+import { StoreGetProductsParamsType } from "./validators"
+
+export async function wrapProductsWithPrices(
+  products,
+  container: MedusaContainer,
+  pricingContext: MedusaPricingContext
+) {
+  if (!isPresent(pricingContext)) {
+    return products
+  }
+
+  const remoteQuery = container.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const pricingModule = container.resolve(ModuleRegistrationName.PRICING)
+  const variants = products.map((product) => product.variants).flat(1)
+
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "product_variant_price_set",
+    fields: ["variant_id", "price_set_id"],
+    variables: {
+      variant_id: variants.map((variantKeys) => variantKeys.id),
+      take: null,
+    },
+  })
+
+  const links = await remoteQuery(queryObject)
+
+  const variantPriceSetMap: Record<string, string> = {}
+  const priceSetCalculatedPriceMap: Record<
+    string,
+    PricingTypes.CalculatedPriceSet[]
+  > = {}
+
+  for (const link of links) {
+    variantPriceSetMap[link.variant_id] = link.price_set_id
+  }
+
+  const calculatedPrices = await pricingModule.calculatePrices(
```

Comment:
I think this is fine, but ideally we can plug this into remote query somehow, so we don't access modules directly in the HTTP layer. Alternatively, should we have this as a workflow? I think for now just leaving a todo comment to reevaluate is fine though

Reply:
I also believe we should have another property to use the remote query.
It can't be `prices` because the object format of calculated prices is different, but we could have a property `calculated_prices` that receives `{context: pricingContext}` as a variable
We could completely eliminate this helper `wrapProductsWithPrices`

Reply:
That sounds good! is it possible to pass deep filters to pricing if product is the entry point to the query?

Reply:
It is, every property that is an entity can have arguments/variables.
however the function `remoteQueryObjectFromString` will have to be adapted to add `__args` in these expansions, not only at the entry point level as it is currently doing.

Reply:
Nice, will take a look. I had this need in a few other places too, so this will be great

Reply:
Should we merge @carlos-r-l-rodrigues's PR and integrate the changes in this PR before we proceed?

Reply:
definitely, can kill a lot of the boiler plate in price list and products